src/recommend.js

```diff
@@ -1,34 +1,55 @@
 import recommendations from '../data/recommendations.json';
+import types from '../data/types.json';
 
 function recommendTreeSpecies(forestType) {
+  if (types.forestType.find(v => v.code === forestType) === undefined) {
+    throw new Error(`${forestType} is not a valid forest Type`);
+  }
   const [, treeSpecies] = Object.entries(recommendations).find(
     t => t[0] === forestType,
   );
   return treeSpecies;
 }
 
-function concatTreeSpecies(treeSp1, treeSp2) {
-  return Array.from(new Set([...treeSp1, ...treeSp2]));
-}
+const concatTreeSpecies = (treeSp1, treeSp2) =>
+  Array.from(new Set([...treeSp1, ...treeSp2]));
 
-function recommend(...forestType) {
-  let result;
-  if (forestType.length === 0) {
+const filterTreeSpecies = (treeSp1, treeSp2) =>
+  treeSp2.filter(tree => !treeSp1.includes(tree));
+
+function recommend(forestType1, forestType2, future) {
+  if (!forestType1) {
     throw new Error(
       `at least one forest type is required to get the recommendation of tree species`,
     );
-  } else if (forestType.length === 1) {
-    result = recommendTreeSpecies(forestType[0]);
-  } else if (forestType.length === 2) {
-    const { one, two, three } = recommendTreeSpecies(forestType[0]);
+  }
+
+  if (future && typeof future !== 'boolean') {
```

Comment:
```suggestion
  if (typeof future !== 'boolean') {
```

Reply:
if there is no check for the "future" then the execution stops at line 27 and throws the error while running other tests.

Reply:
Ah, ok, makes sense.